pkg/operator/controller/ingress/status.go

```diff
@@ -632,21 +633,41 @@ func formatConditions(conditions []*operatorv1.OperatorCondition) string {
 // computeIngressProgressingCondition computes the ingresscontroller's
 // "Progressing" status condition, which indicates if the ingresscontroller's
 // current state matches its desired state.
-func computeIngressProgressingCondition(conditions []operatorv1.OperatorCondition, ic *operatorv1.IngressController, service *corev1.Service, platform *configv1.PlatformStatus) operatorv1.OperatorCondition {
+func computeIngressProgressingCondition(conditions []operatorv1.OperatorCondition, ic *operatorv1.IngressController, service *corev1.Service, platform *configv1.PlatformStatus, deployment *appsv1.Deployment) operatorv1.OperatorCondition {
 	condition := operatorv1.OperatorCondition{
 		Type:   operatorv1.OperatorStatusTypeProgressing,
 		Status: operatorv1.ConditionFalse,
 	}
+	messages := []string{}
+	progressingReasons := []string{}
+
+	// If deployment doesn't have the desired number of replicas, mark as progressing.
+	wantReplicas := int32(0)
+	if deployment.Spec.Replicas != nil {
+		wantReplicas = *deployment.Spec.Replicas
+	}
+	haveReplicas := deployment.Status.Replicas
+	// First check if we have the total number of desired replicas, then check if they are all up-to-date.
+	if haveReplicas != wantReplicas {
+		progressingReasons = append(progressingReasons, "Reconciling")
```

Comment:
I agree this seems funky, but I copied the pattern from https://github.com/openshift/cluster-dns-operator/blob/d50df32df68f53c1d47db8f5e51a8b27c402f278/pkg/operator/controller/dns_status.go#L221. 

It does seem a little awkward in the output:

> NAME      VERSION   AVAILABLE   PROGRESSING   DEGRADED   SINCE   MESSAGE
> ingress   unknown   True        True          False      14m     ingresscontroller "default" is progressing: ReconcilingIncompleteStatus: Have 3 available router pods, want 2....

It only will output the first part of the error message that gets generated first (I put these new errors before than the loadbalancing checks...so always will be this new progressing error).